layers.go

```diff
@@ -1393,18 +1399,30 @@ func (r *layerStore) LayersByUncompressedDigest(d digest.Digest) ([]Layer, error
 
 func (r *layerStore) Lock() {
 	r.lockfile.Lock()
+	if r.rolockfile != nil {
+		r.rolockfile.Lock()
+	}
 }
 
 func (r *layerStore) RecursiveLock() {
 	r.lockfile.RecursiveLock()
+	if r.rolockfile != nil {
+		r.rolockfile.RecursiveLock()
+	}
 }
 
 func (r *layerStore) RLock() {
 	r.lockfile.RLock()
+	if r.rolockfile != nil {
+		r.rolockfile.RLock()
+	}
 }
 
 func (r *layerStore) Unlock() {
 	r.lockfile.Unlock()
+	if r.rolockfile != nil {
+		r.rolockfile.Unlock()
+	}
 }
```

Comment:
@vrothberg @nalind tried another approach with a separate `r.rolockfile` which is additionally set when layer store is locked read and write. I'm not sure if we can solve possible locking issues with that :thinking: